### PR TITLE
JAMES-3775 Rspamd per user reporting

### PR DESCRIPTION
**Pull request by quantranhong1999**
Rebase on master, contribute my work on fixup commits.
Not a code or Rspamd issue, just a Bayes configuration issue.
Result:
```java
  Bob spam score before: 20.825
  Alice spam score before: 20.825
  Bob spam score after: 25.925
  Alice spam score after: 20.825

  Bob ham score before: 0.99
  Alice ham score before: 0.99
  Bob ham score after: 0.374475
  Alice ham score after: 0.99
```

We have the option to enable per-user in Rspamd mailet but not for listener and tasks (always call use per-user bayes). Maybe we should have a configuration option for them?

### CODE REVIEW

File: third-party/rspamd/sample-configuration/statistic.conf [hunk 1]
```diff
@@ -5,20 +5,20 @@ classifier "bayes" {
...
   statfile {
-    symbol = "BAYES_HAM_USER";
+    symbol = "BAYES_HAM";
     spam = false;
   }
   statfile {
-    symbol = "BAYES_SPAM_USER";
+    symbol = "BAYES_SPAM";
     spam = true;
   }
```
Comment (quantranhong1999) [inline third-party/rspamd/sample-configuration/statistic.conf:23]: The main difference makes per-user Bayes work.

File: third-party/rspamd/sample-configuration/statistic.conf [hunk 2]
```diff
@@ -5,20 +5,20 @@ classifier "bayes" {
   cache {
   }
   new_schema = true; # Always use new schema
-  store_tokens = true; # Redefine if storing of tokens is desired
-  signatures = true; # Store learn signatures
+  store_tokens = false; # Redefine if storing of tokens is desired
+  signatures = false; # Store learn signatures
```
Comment (quantranhong1999) [inline third-party/rspamd/sample-configuration/statistic.conf:9]: Better to restore to the default values of Rspamd

**Review by vttranlina — APPROVED** [review]
good job

**Review by chibenwa — APPROVED** [review]

**Comment by quantranhong1999** [comment]
> We have the option to enable per-user in Rspamd mailet but not for listener and tasks (always call use per-user bayes). Maybe we should have a configuration option for them?

I am thinking about a centralized per-user configuration entry for 3 stuff (mailet, listener, tasks) in `rspamd.properties`. I think we should have a simple button to unify them. Imagine if our mailet use global Bayes while tasks/listener enrich per-user Bayes => both Bayes are useless.

WDYT?

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Listener and task do not need extra conf as they could always tell rspamd which user the message belong to, without it hurting.

Ie no need to change conf. No specific conf for this for the listener or the mailet. 

Or did i got smtg wrong?

**Comment by quantranhong1999** [comment]
> No specific conf for this for the listener or the mailet.

I tend to simply our per-user Bayes configuration instead.

In `rspamd.properties`:
```properties
# Whether to scan/learn mails using per-user Bayes. Default to false.
perUserBayes=true
```

**Pull request merged**